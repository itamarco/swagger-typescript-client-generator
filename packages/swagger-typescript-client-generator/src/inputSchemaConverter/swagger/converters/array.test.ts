import { convertSwaggerSchemaToNormalizedSchema } from "./index"
import * as Swagger from "swagger-schema-official"

const testCases: Swagger.Schema[] = [
  {
    type: "array",
    description: "type without items"
  },
  {
    type: "array",
    items: {},
    description: "type with single item"
  }
]
const type = "array"

test.each(testCases.map(item => [item.description, item]))(
  `converter converts scenario "%s"`,
  (description, definition) => {
    const spec = convertSwaggerSchemaToNormalizedSchema(definition as unknown)
    expect(spec.type).toEqual(type)
    expect(spec).toMatchSnapshot()
  }
)
