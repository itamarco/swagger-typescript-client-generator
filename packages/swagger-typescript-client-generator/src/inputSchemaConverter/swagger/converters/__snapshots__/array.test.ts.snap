// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`converter converts scenario "type with single item" 1`] = `
Object {
  "description": "type with single item",
  "items": Object {
    "description": undefined,
    "type": "unknown",
  },
  "type": "array",
}
`;

exports[`converter converts scenario "type without items" 1`] = `
Object {
  "description": "type without items",
  "items": Object {
    "type": "unknown",
  },
  "type": "array",
}
`;
